pandas/_libs/index.pyx

```diff
@@ -8,10 +8,14 @@ from cpython.slice cimport PySlice_Check
 
 import numpy as np
 cimport numpy as cnp
-from numpy cimport (ndarray, float64_t, int32_t, int64_t, uint8_t, uint64_t,
-                    NPY_DATETIME, NPY_TIMEDELTA)
+from numpy cimport ndarray, float64_t, int32_t, int64_t, uint8_t, uint64_t
 cnp.import_array()
 
+cdef extern from "numpy/arrayobject.h":
+    # These can be cimported directly from numpy in cython>=0.27.3
```

Comment:
should prob put array dtype introspection in a single place and expose via``is_timedelta64_array`` (new) somewhere maybe util.pxd